### PR TITLE
DUPLO-14760 Mandatory field "Root Password" for database creation is missing for Cloud SQL databases.

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Add support for root password
## Summary of changes
added support for password for gcp sql resource
This PR does the following:

- Added support for root password passing
- ...

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [✓ ] Manually, on a remote test system

## Describe any breaking changes

- ...

___

## **Type**
enhancement, documentation

___

## **Description**
- Added support for specifying a root password for GCP SQL database instances, enhancing security and customization.
- Implemented logic to require a root password for certain SQL Server versions, improving compliance with best practices.
- Updated SDK to support root password management for GCP SQL databases.
- Updated documentation and changelog to reflect the new feature and its usage.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_gcp_sql_database.go</strong><dd><code>Add Root Password Support for GCP SQL Database Instances</code>&nbsp; </dd></summary>
<hr>
      
duplocloud/resource_duplo_gcp_sql_database.go

<li>Added <code>root_password</code> field to the GCP SQL database schema.<br> <li> Implemented <code>checkPasswordNeeded</code> function to determine if a root <br>password is required based on the database version.<br> <li> Enforced root password requirement during resource creation if <br>applicable.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/475/files#diff-5e2b59a959de92a468b84aa15f4066eb662d23e4c9a0e8ef04fca002d5235831">+27/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_sql_database.go</strong><dd><code>SDK Support for GCP SQL Database Root Password</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
duplosdk/gcp_sql_database.go

<li>Added <code>RootPassword</code> field to <code>DuploGCPSqlDBInstance</code> struct to support <br>root password management.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/475/files#diff-ef4280fb2cddb49187c078fc9a79d10c1b8a94e27286a2e4c718da2d320ac2d8">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update Changelog for Root Password Support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
CHANGELOG.md

<li>Updated the changelog to include the addition of root password support <br>for GCP SQL database instances.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/475/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+16/-15</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>gcp_sql_database_instance.md</strong><dd><code>Document Root Password Parameter for GCP SQL Database Instances</code></dd></summary>
<hr>
      
docs/resources/gcp_sql_database_instance.md

<li>Documented the <code>root_password</code> optional parameter for GCP SQL database <br>instances.<br>

</details>
    

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/475/files#diff-5695da959b1954953ab1c73c64b4a4fc96c884d73e08e5119eb6b6b67d008b6f">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-14760 TF:GCP: Mandatory field "Root Password" for database creation is missing for Cloud SQL databases.](https://app.clickup.com/t/8655600/DUPLO-14760)

**Review by michalpleszkun — DISMISSED** [review]

**Review by komalkadam-dc — APPROVED** [review]

File: duplocloud/resource_duplo_gcp_sql_database.go [hunk 1]
```diff
@@ -77,7 +77,30 @@ func gcpSqlDBInstanceSchema() map[string]*schema.Schema {
 			Optional:    true,
 			Default:     true,
 		},
+		"root_password": {
+			Description: "root password for specific database versions",
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_gcp_sql_database.go:81]: Change this to "Provide root password for specific database versions."

Comment (tahir-duplo) [reply duplocloud/resource_duplo_gcp_sql_database.go:81]: Add computed as well.

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_gcp_sql_database.go]: done

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**